admin/addLink.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import * as firebase from 'firebase/app';
+
+type Data = {
+    title: string,
+    url: string,
+    tags: Array<string>
+}
+
+export const AddLink = () => {
+
+    const [data, setData] = useState({title: "", url: "", tags: [""]});
+
+    const handleChange = (e :React.ChangeEvent<HTMLInputElement>) => {
+        switch(e.target.name){
+            case 'title':
+                setData({...data, title: e.target.value})
+                break;
+            case 'url':
+                setData({...data, url: e.target.value})
+                break;
+            case 'tags':
+                const tags = e.target.value.split(", ")
+                setData({...data, tags: tags})
+                break;
+        } 
+    }
```

Comment:
Dette fungerer fint, men siden dataen aldri endrer seg samtidig, så ville jeg bare hatt tre state-variabler istedenfor:

```js
const [title, setTitle] = React.useState('');
const [url, setUrl] = React.useState('');
const [tags, setTags] = React.useState('');
```

og hatt tre forskjellige change handlers - men de kan du bare spesifisere inline:

```js
<input value={title} onChange={e => setTitle(e.target.value)} />
```

osv.

Så kan du endre `tags` til et array i `sendData`-metoden, hvor du setter sammen alt:

```js
const sendData = (e: React.FormEvent) => {
  e.preventDefault();
  addToDatabase({ title, url, tags: tags.split(', ');
};
```

Reply:
Lurt. Takk! Spørsmål: hvorfor slipper jeg å skrive: `addToDatabase({ title: title, url: url...`?

Reply:
Godt spørsmål! Det er en feature i JavaScript at når navn på nøkkelen i objektet og variabelen på verdien du gir den nøkkelen er det samme, så kan du bare skrive det en gang, for å slippe repetisjon.

Dvs at 

```js
const firstName = ‘stian’;
const person = { firstName: firstName };
```

Kan skrives 

```js
const firstName = ‘stian’;
const person = { firstName };
```